tests/test_userspace.py

```diff
@@ -75,4 +75,15 @@ def test_hostname(shell):
 
     [hostname] = shell.run_check("hostname")
 
-    assert serial_number in hostname
+    with check:
+        assert serial_number in hostname
+
+    [etc_hostname] = shell.run_check("cat /etc/hostname")
+
+    with check:
+        assert etc_hostname != "localhost"
+
+    with check:
+        assert serial_number in etc_hostname
+
+
```

Comment:
The localhost test is of cause redundant (the serial number test would trigger as well), but it does show the intent of introducing the change very well, so I think we should keep it.

I would prefer if you could remove the trailing newlines though …

Reply:
Ouch. Moved the trailing newlines into the correct commit.